### ISSUE TITLE
My own data training does not converge

### ISSUE THREAD

**Issue by ximiluuuu** [Physical-Intelligence/openpi#432]
I collected over 300 sets(Each set of data takes approximately 30 seconds) of data and trained 30000 times. When converging, the loss was about 2.5+, and the loss of the same dataset, ACT' loss was within 0.01,Where exactly is the problem？

![Image](https://github.com/user-attachments/assets/e44af44b-e385-446c-a540-11c08aa75bad)

```
    TrainConfig(
        name="pi0_fast_base_libero_lora",
        model=pi0_fast.Pi0FASTConfig(action_dim=7, action_horizon=10, max_token_len=180, paligemma_variant="gemma_2b_lora"),
        data=LeRobotLiberoDataConfig(
            repo_id="merge_data",
            base_config=DataConfig(
                local_files_only=True,
                prompt_from_task=True,
            ),
        ),
        batch_size=1,
        freeze_filter=pi0_fast.Pi0FASTConfig(
            action_dim=7, action_horizon=10, max_token_len=180, paligemma_variant="gemma_2b_lora"
        ).get_freeze_filter(),
        weight_loader=weight_loaders.CheckpointWeightLoader("s3://openpi-assets/checkpoints/pi0_base/params"),
        num_train_steps=30000,
        fsdp_devices=1,
        ema_decay=None,
    )
```

**Comment by kpertsch** [comment]
Hi!
Thanks for trying out our models! Please note that ACT and the pi0-FAST model you are using have very different training objectives and loss functions (ACT uses L1 loss I believe, while pi0-FAST uses a discrete cross-entropy loss over action tokens). As such, it is not surprising that the loss magnitudes are very different and this does not indicate that the pi0-FAST model is not converged. I would suggest looking at the learning curve of pi0-FAST and if it looks converged then you can go ahead and try running the model. 

I hope that makes things a bit clearer! I will close this issue for now, but feel free to reopen!
